Gymphony/src/app/core/components/profile-icon/profile-icon.component.ts

```diff
@@ -0,0 +1,24 @@
+import { Component, Input, Output, EventEmitter } from '@angular/core';
+import { User } from '../../interfaces/user';
+import { ModalService } from '../../../features/auth/services/modal.service';
+import { AuthService } from '../../../features/auth/services/auth.service';
+
+@Component({
+  selector: 'app-profile-icon',
+  templateUrl: './profile-icon.component.html',
+  styleUrl: './profile-icon.component.css'
+})
+export class ProfileIconComponent {
+
+  constructor(private modalService: ModalService, private authService: AuthService) { }
+  
+  @Input() user?: User;
```

Comment:
Let's put component variables in the following order:
Inputs
Outputs
internal variables
constructor
ngOnInit if present
public methods
protected methods
private methods